### ISSUE TITLE
FNX3-22103 ⁃ Gradle: Cannot change attributes of configuration after it has been resolved

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/fenix#13527]
With Android Gradle plugin 3.2.1 I get the following error:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/skaspari/Projects/Mozilla/android-components/build.gradle' line: 160

* What went wrong:
A problem occurred evaluating root project 'android-components'.
> A problem occurred configuring project ':service-firefox-accounts'.
   > Cannot change attributes of configuration ':service-firefox-accounts:kapt' after it has been resolved
```

That's where we call:

```
evaluationDependsOnChildren()
```

Afaik we had to add this for dokka. Once we update the plugin we may need to re-visit that part of the script.

**Comment by data-sync-user** [comment]
➤ Nick Alexander commented:

I think the kapt plugin is just badly written; I see this when adding any configuration over in a-s.  Others see the same.

**Comment by data-sync-user** [comment]
➤ Sebastian Kaspari commented:

Afaik we had saw the same issue when updating Kotlin to 1.3.0 and @csadilek debugged this and found a workaround. Maybe this is now unblocked?

**Comment by data-sync-user** [comment]
➤ Christian Sadilek commented:

Unblocked for us, but not a-s, unfortunately. The problem was triggered by the jnaForTest workaround: https://github.com/mozilla/application-services/blob/cf4e84f84f51f55ab9eecd2ed8389fd532e9a008/components/places/android/library/build.gradle#L84

We used .files which resolved dependencies directly. So, basically this line:
testImplementation files(configurations.jnaForTest.files)

It's (temporarily) unblocked for us, as all JNA tests moved to a-s. I just spent a few more hours looking for an alternative workaround that will work despite the kapt plugin issue, but no luck so far.
